test/specs/column/tests/01.columns_list.js

```diff
@@ -757,6 +760,88 @@ exports.execute = function (options) {
                 areSameColumnList(compactRef.generateColumnsList(), compactColumns);
                 areSameColumnList(entryEditRef.generateColumnsList(), entryEditRef.columns);
             });
+
+            describe('with system columns heuristic config options defined,', function () {
+                function systemColumnsHeuristicsMode(context) {
+                    var mode = null;
+                    if (context == 'compact') {
+                        mode = true;
+                    } else if (context == 'detailed') {
+                        mode = ['RMB', 'RCT'];
```

Comment:
You should test an array that is rearranged. To make sure we're preserving the order.

Reply:
Also add RID, a non-system column, and an invalid column to this list. To make sure those branching logic that you have in the code are working as expected.